llms/mlx_lm/utils.py

```diff
@@ -384,6 +388,26 @@ def save_weights(save_path: Union[str, Path], weights: Dict[str, Any]) -> None:
         else "model.safetensors"
     )
 
+    index_data = {"metadata": {"total_size": 0}, "weight_map": {}}
+
     for i, shard in enumerate(shards):
         shard_name = shard_file_format.format(i + 1, shards_count)
-        mx.save_safetensors(str(save_path / shard_name), shard)
+        shard_path = save_path / shard_name
+
+        mx.save_safetensors(str(shard_path), shard)
+
+        for weight_name in shard.keys():
+            index_data["weight_map"][weight_name] = shard_name
+
+        index_data["metadata"]["total_size"] += shard_path.stat().st_size
```

Comment:
The total size is supposed to be the size in bytes right? (As compared to say the # of params or something)?

Reply:
I just implemented the naive way to put the file size in bytes therre. If you prefer, I can double-check the transformers implementation to see if it should be "pramas size" or "file size".

Reply:
Maybe @pcuenca knows. @pcuenca we are wondering if in safetensors index json, the `total_size` is bytes or # params or some other thing. Do you happen to know? (E.g. here https://huggingface.co/mistralai/Mistral-7B-v0.1/blob/main/model.safetensors.index.json#L3)

Reply:
Looks like it's the parameter size. I will update the implementation accordingly.
https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L379

Reply:
Thanks!!